src/outputs/gadget.js

```diff
@@ -33,8 +33,9 @@ import languageBlob from '../../temp/languages'; // This is generated during the
 
 		loadWhoWroteThat();
 
-		if ( welcomeTourSeen ) {
+		if ( welcomeTourSeen || !activationInstance.isValidPage() ) {
```

Comment:
I'm sure you're right here, but I can't quite figure it out myself. I would have said that this should be

    if ( welcomeTourSeen || !$( '#t-whowrotethat' ).length ) {

to be equivalent to what's done in `src/outputs/browserextension_tour.js`. (And that maybe the check in that file could happen in a similar place as this one, i.e. *before* loading the required modules... but that's probably a separate matter.)

Reply:
That's a good point, although it's also basically the exact same, since `$( '#t-whowrotethat' )` would only be injected to the DOM if `isValidPage` is true, so it's more a logical/flow thing.

I can switch it up for logic's sake.